### PR TITLE
Evaluation algorithm for v2

### PR DESCRIPTION
**Pull request by jkodumal**

### CODE REVIEW

File: test/test.js [hunk 1]
```diff
@@ -1,180 +0,0 @@
-var rewire = require('rewire');
```
Comment (jkodumal) [inline test/test.js]: This is all covered by the integration harness now.

**Comment by jkodumal** [comment]
You can ignore the test failures-- I deleted the old tests and will be re-writing new ones. However, this does pass the new v2 integration harness tests.

File: evaluate_flag.js [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+              callback(new Error("Unsatisfied prerequisite"), null)
+            } else { 
+              // The prerequisite was satisfied
+              callback(null, null);
+            }
+          });          
+        });
+      }, 
+      function(err, results) {
+        var i;
```
Comment (drichelson) [inline evaluate_flag.js]: i is unused.

File: evaluate_flag.js [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+    target = flag.targets[i];
+
+    if (!target.values) {
+      continue;
+    }
+
+    for (j = 0; j < target.values.length; j++) {
+      if (user.key === target.values[j]) {
+        variation = get_variation(flag, target.variation);
+        cb(variation === null ? new Error("Undefined variation") : null, variation);
```
Comment (drichelson) [inline evaluate_flag.js]: I think we should add a little more context to these data validation errors. If they pop up we'll be able to get some value if we include at least the feature key.

File: requestor.js [hunk 4]
```diff
@@ -30,9 +30,7 @@ function Requestor(api_key, config) {
   });
 
   function resource_path(latest) {
-    return latest ? '/api/eval/latest-features' : '/api/eval/features';
-   // TODO Update to the following v2 paths: 
-   // return latest ? '/sdk/latest-flags': '/sdk/flags';
+    return latest ? '/sdk/latest-flags': '/sdk/flags';
```
Comment (drichelson) [inline requestor.js]: Let's go ahead and just always return '/sdk/latest-flags' here since we no longer use the non-latest endpoint.

**Comment by drichelson** [comment]
👍 Once my minor issues are addressed.

**Comment by drichelson** [comment]
👍

**Comment by jkodumal** [comment]
I'll update the tests in a separate PR.

**Pull request merged**